code/lib/create-storybook/src/generators/REACT_NATIVE/index.ts

```diff
@@ -1,71 +1,104 @@
-  });
-};
-
-export default generator;
+import {
+  ProjectType,
+  SupportedLanguage,
+  copyTemplateFiles,
+  getBabelDependencies,
+} from 'storybook/internal/cli';
+import { CLI_COLORS, logger } from 'storybook/internal/node-logger';
+import { SupportedBuilder, SupportedRenderer } from 'storybook/internal/types';
+
+import { dedent } from 'ts-dedent';
+
+import { defineGeneratorModule } from '../modules/GeneratorModule';
+
+export default defineGeneratorModule({
+  metadata: {
+    projectType: ProjectType.REACT_NATIVE,
+    renderer: SupportedRenderer.REACT,
+    builderOverride: SupportedBuilder.WEBPACK5,
+    framework: null,
+  },
+  configure: async (packageManager, context) => {
+    const missingReactDom = !packageManager.getDependencyVersion('react-dom');
+    const reactVersion = packageManager.getDependencyVersion('react');
+    const dependencyCollector = context.dependencyCollector;
+
+    const peerDependencies = [
+      'react-native-safe-area-context',
+      '@react-native-async-storage/async-storage',
+      '@react-native-community/datetimepicker',
+      '@react-native-community/slider',
+      'react-native-reanimated',
+      'react-native-gesture-handler',
+      '@gorhom/bottom-sheet',
+      'react-native-svg',
+    ].filter((dep) => !packageManager.getDependencyVersion(dep));
+
+    const packagesToResolve = [
+      ...peerDependencies,
+      '@storybook/addon-ondevice-controls',
+      '@storybook/addon-ondevice-actions',
+      '@storybook/react-native',
+      'storybook',
+    ];
+
+    const versionedPackages = await packageManager.getVersionedPackages(packagesToResolve);
+    const babelDependencies = await getBabelDependencies(packageManager as any);
+
+    const packages: string[] = [
+      ...babelDependencies,
+      ...versionedPackages,
+      ...(missingReactDom && reactVersion ? [`react-dom@${reactVersion}`] : []),
+    ];
+
+    dependencyCollector.addDependencies(packages);
+
+    // Add React Native specific scripts
+    packageManager.addScripts({
+      'storybook-generate': 'sb-rn-get-stories',
+    });
+
+    const storybookConfigFolder = '.rnstorybook';
+
+    // Copy React Native templates
+    await copyTemplateFiles({
+      packageManager: packageManager as any,
+      templateLocation: SupportedRenderer.REACT_NATIVE,
+      language: SupportedLanguage.TYPESCRIPT,
+      destination: storybookConfigFolder,
+      features: context.features,
+    });
+
+    // React Native doesn't use baseGenerator - return special config
+    return {
+      // Signal to skip baseGenerator by returning minimal config
+      storybookConfigFolder,
+      skipGenerator: true,
+      storybookCommand: null,
+      shouldRunDev: false, // React Native needs additional manual steps to configure the project
+    };
+  },
+  postConfigure: ({ packageManager }) => {
+    logger.log(dedent`
+      ${CLI_COLORS.warning('React Native (RN) Storybook installation is not 100% automated.')}
```

Comment:
I think we call it "Storybook for React Native" usually. Also I would just spell it out in full each time rather than use an acronym.